framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -528,56 +579,152 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
                 }
             }
         }
+
+        // Refine the values of the operands in the store based on the result of the comparison.
+
+        List<Number> thenLeftVals = new ArrayList<>();
+        List<Number> elseLeftVals = new ArrayList<>();
+
+        // Because resultValues contains the (boolean) results of each comparison (i.e. the product of the number
+        // of elements in each list), we have to iterate through parts of the list at a time in order to find all
+        // the results of comparisons involving a particular value in the left list.
+        for (int i = 0; i < lefts.size(); i++) {
+            for (int j = i; j < (1 + i) * rights.size(); j++) {
```

Comment:
It's really hard to understand why `(1 + i( * rights.size()` is correct.  Could this logic (the logic for the rightValues)  be moved into the nested for loop above?